db/external_sst_file_ingestion_job.h

```diff
@@ -151,6 +159,20 @@ class ExternalSstFileIngestionJob {
   int ConsumedSequenceNumbersCount() const { return consumed_seqno_count_; }
 
  private:
+  Status ResetTableReader(const std::string& external_file,
+                          uint64_t new_file_number, SuperVersion* sv,
+                          IngestedFileInfo* file_to_ingest,
+                          std::unique_ptr<TableReader>* table_reader);
+
+  // Read the external file's table properties to do various sanity checks.
+  // In some cases when sanity check passes, `table_reader` could be reset with
+  // different options. For example: when external file does not contain
+  // timestamps while column family enables UDT in Memtables only feature.
+  Status SanityCheckTableProperties(const std::string& external_file,
```

Comment:
It also initializes a couple fields in `IngestedFileInfo` that is not obvious from the name "SanityCheck" now.

Reply:
Thank you for the suggestion! It's however not straightforward to rename the function to point out this fact. So I added some documentation for this, specifically, it's this part: "...to do various sanity checks _and populates certain fields in `IngestedFileInfo` according to some table properties_"